File: next-tauri/src/platform/contracts/image.ts
export interface ImageAssetSummary {
  name: string
  mimeType: string
  width: number
  height: number
  sizeBytes: number
  updatedAt: number
}

export interface ImageAsset extends ImageAssetSummary {
  dataUrl: string
}

export interface ImageAssetInput {
  name: string
  dataUrl: string
  width: number
  height: number
}

export type ImageVectorizePreset = 'poster' | 'photo' | 'bw'
export type ImageVectorizeDetail = 'low' | 'medium' | 'high'

export interface ImageVectorizeOptions {
  preset: ImageVectorizePreset
  colorCount: number
  detail: ImageVectorizeDetail
  filterSpeckle: number
}

export interface ImageApi {
  list(): Promise<ImageAssetSummary[]>
  save(input: ImageAssetInput): Promise<ImageAssetSummary>
  importPaths(paths: string[]): Promise<ImageAssetSummary[]>
  read(name: string): Promise<ImageAsset>
  export(names: string[]): Promise<string[] | null>
  vectorize(names: string[], options: ImageVectorizeOptions): Promise<string[] | null>
  rename(name: string, nextName: string): Promise<ImageAssetSummary>
  delete(names: string[]): Promise<number>
}

File: next-tauri/src-tauri/src/contracts/image.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ImageAssetInput {
    pub name: String,
    pub data_url: String,
    pub width: u32,
    pub height: u32,
}

#[derive(Clone, Debug, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ImageAssetSummary {
    pub name: String,
    pub mime_type: String,
    pub width: u32,
    pub height: u32,
    pub size_bytes: usize,
    pub updated_at: i64,
}

#[derive(Clone, Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ImageAsset {
    #[serde(flatten)]
    pub summary: ImageAssetSummary,
    pub data_url: String,
}

#[derive(Clone, Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ImageVectorizeOptions {
    pub preset: String,
    pub color_count: u8,
    pub detail: String,
    pub filter_speckle: usize,
}
